pkg/registry/service/rest.go

```diff
@@ -233,19 +234,43 @@ var _ = rest.Redirector(&REST{})
 
 // ResourceLocation returns a URL to which one can send traffic for the specified service.
 func (rs *REST) ResourceLocation(ctx api.Context, id string) (*url.URL, http.RoundTripper, error) {
-	eps, err := rs.endpoints.GetEndpoints(ctx, id)
+	// Allow ID as "svcname" or "svcname:port".
+	parts := strings.Split(id, ":")
+	if len(parts) > 2 {
+		return nil, nil, errors.NewBadRequest(fmt.Sprintf("invalid service request %q", id))
+	}
+	svcName := parts[0]
+	portStr := ""
+	if len(parts) == 2 {
+		portStr = parts[1]
+	}
+
```

Comment:
To me, it only makes sense to permit the port to be omitted in the case that the service exposes just one port. Otherwise, we should require servicename:portnumber or servicename:portname.